tests/compile-fail/unsupported_posix_dlsym.rs

```diff
@@ -0,0 +1,14 @@
+// ignore-windows: No dlsym() on Windows
+
+#![feature(rustc_private)]
+
+extern crate libc;
+
+use std::ptr;
+
+fn main() {
+    unsafe {
+        libc::dlsym(ptr::null_mut(), b"foo\0".as_ptr().cast());
+        //~^ ERROR unsupported operation: unsupported
+    }
+}
```

Comment:
I'm not sure if my changes will affect `dlsym` or not, so I wrote this (and the `unsupported_windows_dlsym.rs`) test.

Reply:
Why specifically `dlsym`? Here you are not *calling* a dlsym'd function, which is the situation where things get special. So this tests seems meaningless to me.

Reply:
Indeed... Maybe I got confused. Removed.